gnovm/pkg/gnolang/type_check.go

```diff
@@ -215,6 +215,152 @@ func assertAssignableTo(n Node, xt, dt Type, autoNative bool) {
 	}
 }
 
+func assertValidConstExpr(store Store, last BlockNode, n *ValueDecl, expr Expr) {
+	if n.Type != nil {
+		nt := evalStaticType(store, last, n.Type)
+		if xnt, ok := nt.(*NativeType); ok {
+			nt = go2GnoBaseType(xnt.Type)
+		}
+
+		if _, ok := baseOf(nt).(PrimitiveType); !ok {
+			panic(fmt.Sprintf("invalid constant type %s", nt.String()))
+		}
+	}
+
+	nt := evalStaticTypeOf(store, last, expr)
+	if xnt, ok := nt.(*NativeType); ok {
+		nt = go2GnoBaseType(xnt.Type)
+	}
+
+	if nt == nil {
+		panic(fmt.Sprintf("%s (variable of type nil) is not constant", expr))
+	}
+
+	if _, ok := baseOf(nt).(PrimitiveType); !ok {
+		panic(fmt.Sprintf("%s (variable of type %s) is not constant", expr, nt))
+	}
+
+	assertValidConstValue(store, last, expr)
+}
+
+func assertValidConstValue(store Store, last BlockNode, currExpr Expr) {
+Main:
+	switch currExpr := currExpr.(type) {
+	case *ConstExpr:
+	case *BasicLitExpr:
```

Comment:
I believe this is not needed, as it's already evaluated to be ConstExpr.

Reply:
right I removed ^^